contracts/Liquidator.sol

```diff
@@ -54,13 +94,43 @@ contract Liquidator is WithAdmin, ReentrancyGuard {
         ensureNonzeroAddress(comptroller_);
         ensureNonzeroAddress(vaiController_);
         ensureNonzeroAddress(treasury_);
-        vBnb = VBNB(vBnb_);
+        vBnb = IVBNB(vBnb_);
         comptroller = IComptroller(comptroller_);
-        vaiController = VAIControllerInterface(vaiController_);
+        vaiController = IVAIController(vaiController_);
         treasury = treasury_;
         treasuryPercentMantissa = treasuryPercentMantissa_;
     }
 
+    /// @notice An admin function to restrict liquidations to allowed addresses only.
+    /// @dev Use {addTo,removeFrom}Allowlist to configure the allowed addresses.
+    /// @param borrower The address of the borrower
+    function restrictLiquidation(address borrower) external onlyAdmin {
+        require(!liquidationRestricted[borrower], "already restricted");
+        liquidationRestricted[borrower] = true;
+        emit LiquidationRestricted(borrower);
+    }
+
+    /// @notice An admin function to remove restrictions for liquidations.
+    /// @dev Does not impact the allowlist for the borrower, just turns off the check.
+    /// @param borrower The address of the borrower
+    function unrestrictLiquidation(address borrower) external onlyAdmin {
+        require(liquidationRestricted[borrower], "not restricted");
+        liquidationRestricted[borrower] = false;
+        emit LiquidationRestrictionsDisabled(borrower);
+    }
+
+    function addToAllowlist(address borrower, address liquidator) external onlyAdmin {
```

Comment:
can we add natspec to this and `removeFromAllowlist` function?

Reply:
Fixed in e39bf65